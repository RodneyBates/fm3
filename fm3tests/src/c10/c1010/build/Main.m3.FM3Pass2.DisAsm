 ItkEOF()
 ItkLeftEnd()
 ItkModuleLt(L2(16_2),[11,0])
 ItkDeclScopeLt(L1(16_1))
 ItkOpenScopeLt(L1(16_1))
 ItkOpenDeclListLt(L11(16_b),[13,2])
 ItkTypeDeclLt([14,7])
 ItkDeclNo(D3,[14,7])
 ItkTypeDeclType([14,9])
 ItkDefTopExprNo(L17(16_11),[14,11])
 ItkTypeDeclRt([14,7])
 ItkOpenDeclListSep(L1(16_1),[13,2])
 ItkTypeDeclLt([15,7])
 ItkDeclNo(D4,[15,7])
 ItkTypeDeclType([15,9])
 ItkDefTopExprNo(L16(16_10),[15,11])
 ItkTypeDeclRt([15,7])
 ItkOpenDeclListSep(L2(16_2),[13,2])
 ItkTypeDeclLt([17,7])
 ItkDeclNo(D5,[17,7])
 ItkTypeDeclType([17,10])
 ItkDefTopExprNo(L15(16_f),[17,12])
 ItkTypeDeclRt([17,7])
 ItkOpenDeclListSep(L3(16_3),[13,2])
 ItkTypeDeclLt([18,7])
 ItkDeclNo(D6,[18,7])
 ItkTypeDeclType([18,10])
 ItkDefTopExprNo(L14(16_e),[18,12])
 ItkTypeDeclRt([18,7])
 ItkOpenDeclListSep(L4(16_4),[13,2])
 ItkTypeDeclLt([19,7])
 ItkDeclNo(D7,[19,7])
 ItkTypeDeclType([19,10])
 ItkDefTopExprNo(L13(16_d),[19,12])
 ItkTypeDeclRt([19,7])
 ItkOpenDeclListSep(L5(16_5),[13,2])
 ItkFullRevealLt([21,9])
 ItkFullRevealType([21,11])
 ItkDefTopExprNo(L11(16_b),[65535,65535])
 ItkFullRevealRt([21,9])
 ItkOpenDeclListSep(L6(16_6),[13,2])
 ItkPartialRevealLt([22,9])
 ItkPartialRevealSubtype([22,16])
 ItkDefTopExprNo(L8(16_8),[22,19])
 ItkPartialRevealRt([22,9])
 ItkOpenDeclListSep(L7(16_7),[13,2])
 ItkPartialRevealLt([23,9])
 ItkPartialRevealSubtype([23,11])
 ItkDefTopExprNo(L6(16_6),[65535,65535])
 ItkPartialRevealRt([23,9])
 ItkOpenDeclListSep(L8(16_8),[13,2])
 ItkFullRevealLt([26,4])
 ItkFullRevealType([26,7])
 ItkObjTypeLt([26,9])
 ItkSupertypeAbsent([26,9])
 ItkDefTopExprNo(L4(16_4),[26,9])
 ItkDeclScopeLt(L2(16_2))
 ItkFieldDeclListLt(L0(16_0),[26,16])
 ItkFieldDeclListRt(L0(16_0),[26,16])
 ItkMethodDeclListLt(L0(16_0),[26,16])
 ItkMethodDeclListRt(L0(16_0),[26,16])
 ItkOverrideListLt(L0(16_0),[26,16])
 ItkOverrideListRt(L0(16_0),[26,16])
 ItkDeclScopeRt(L2(16_2))
 ItkObjTypeRt([26,9])
 ItkFullRevealRt([26,4])
 ItkOpenDeclListSep(L1(16_1),[27,2])
 ItkPartialRevealLt([27,4])
 ItkPartialRevealSubtype([27,7])
 ItkPartialRevealRt([27,4])
 ItkOpenDeclListSep(L2(16_2),[28,2])
 ItkFullRevealLt([28,4])
 ItkFullRevealType([28,7])
 ItkDefTopExprNo(L1(16_1),[28,9])
 ItkFullRevealRt([28,4])
 ItkOpenDeclListRt(L11(16_b),[13,2])
 ItkModuleBegin(L2(16_2),[30,2])
 ItkStmtListLt(L0(16_0),[31,2])
 ItkStmtListRt(L0(16_0),[31,2])
 ItkOpenScopeRt(L1(16_1))
 ItkDeclScopeRt(L1(16_1))
 ItkModuleRt(L2(16_2),[11,0])
 ItkRightEnd()
 ItkBOF()
     0
