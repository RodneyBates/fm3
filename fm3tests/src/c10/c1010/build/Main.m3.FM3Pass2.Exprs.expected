From unit expression map, Expr No 0 NIL

From unit expression map, Expr No 1 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 1
 ExpPosition = [28,9]
 ExpOpcode = NULL
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 2 16_****************
 ExpStackLink = NIL
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 2
 ExpPosition = [28,4]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = reference
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 7


From unit expression map, Expr No 3 16_****************
 ExpStackLink = ExprNo 2 at 16_**************** [28,4]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 3
 ExpPosition = [27,4]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = reference
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 6


From unit expression map, Expr No 4 16_****************
 ExpStackLink = ExprNo 3 at 16_**************** [27,4]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 4
 ExpPosition = [26,9]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = <unknown>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = FALSE
 ExpRefTypeIsUntraced = FALSE


From unit expression map, Expr No 5 16_****************
 ExpStackLink = ExprNo 3 at 16_**************** [27,4]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 5
 ExpPosition = [26,4]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = reference
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 5


From unit expression map, Expr No 6 16_****************
 ExpStackLink = ExprNo 5 at 16_**************** [26,4]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 6
 ExpPosition = [65535,65535]
 ExpOpcode = %UNTRACED_ROOT
 ExpDownKind = <null>
 ExpUpKind = type
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 7 16_****************
 ExpStackLink = ExprNo 5 at 16_**************** [26,4]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 7
 ExpPosition = [23,9]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = reference
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprIdentRefTyp: *)
 ExpIdentDeclNo = 4


From unit expression map, Expr No 8 16_****************
 ExpStackLink = ExprNo 7 at 16_**************** [23,9]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 8
 ExpPosition = [22,19]
 ExpOpcode = REFANY
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 9 16_****************
 ExpStackLink = ExprNo 7 at 16_**************** [23,9]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 9
 ExpPosition = [22,13]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = reference
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype Expr1OpndTyp: *)
 ExpOpnd1 = ExprNo 10 16_****************
.. ExpStackLink = ExprNo 9 at 16_**************** [22,13]
.. ExpType = NIL
.. ExpRefConstVal = 16_****************
.. ExpScalarConstVal = 0
.. ExpLoTypeInfoRef = 
.. ExpReachedDeclNoSet = {}
.. ExpSelfExprNo = 10
.. ExpPosition = [22,9]
.. ExpOpcode = <RidNull>
.. ExpDownKind = <null>
.. ExpUpKind = reference
.. ExpKind = <null>
.. ExpState = EsUnresolved
.. ExpIsConst = FALSE
.. ExpConstValIsKnown = FALSE
.. ExpIsUsable = TRUE
.. ExpIsLegalRecursive = FALSE
.. ExpIsDesignator = FALSE
.. ExpIsWritable = FALSE
.. ExpIsPresent = TRUE
.. ExpRefTypeIsUntraced = FALSE
.. (* From subtype ExprIdentRefTyp: *)
.. ExpIdentDeclNo = 3

 (* From subtype ExprDotTyp: *)
 ExpDotIdAtom = X


From unit expression map, Expr No 11 16_****************
 ExpStackLink = ExprNo 9 at 16_**************** [22,13]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 11
 ExpPosition = [65535,65535]
 ExpOpcode = %ROOT
 ExpDownKind = <null>
 ExpUpKind = type
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 12 16_****************
 ExpStackLink = ExprNo 9 at 16_**************** [22,13]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 12
 ExpPosition = [21,9]
 ExpOpcode = <RidNull>
 ExpDownKind = <null>
 ExpUpKind = reference
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = FALSE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprRemoteRefTyp: *)
 ExpRemoteUnitNo = 1
 ExpRemoteDeclNo = 1


From unit expression map, Expr No 13 16_****************
 ExpStackLink = ExprNo 12 at 16_**************** [21,9]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 13
 ExpPosition = [19,12]
 ExpOpcode = BOOLEAN
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 14 16_****************
 ExpStackLink = ExprNo 12 at 16_**************** [21,9]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 14
 ExpPosition = [18,12]
 ExpOpcode = WIDECHAR
 ExpDownKind = <null>
 ExpUpKind = type
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 15 16_****************
 ExpStackLink = ExprNo 12 at 16_**************** [21,9]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 15
 ExpPosition = [17,12]
 ExpOpcode = CHAR
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 16 16_****************
 ExpStackLink = ExprNo 12 at 16_**************** [21,9]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 16
 ExpPosition = [15,11]
 ExpOpcode = BOOLEAN
 ExpDownKind = <null>
 ExpUpKind = <null>
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


From unit expression map, Expr No 17 16_****************
 ExpStackLink = ExprNo 12 at 16_**************** [21,9]
 ExpType = NIL
 ExpRefConstVal = 16_****************
 ExpScalarConstVal = 0
 ExpLoTypeInfoRef = 
 ExpReachedDeclNoSet = {}
 ExpSelfExprNo = 17
 ExpPosition = [14,11]
 ExpOpcode = WIDECHAR
 ExpDownKind = <null>
 ExpUpKind = type
 ExpKind = <null>
 ExpState = EsUnresolved
 ExpIsConst = FALSE
 ExpConstValIsKnown = FALSE
 ExpIsUsable = TRUE
 ExpIsLegalRecursive = TRUE
 ExpIsDesignator = FALSE
 ExpIsWritable = FALSE
 ExpIsPresent = TRUE
 ExpRefTypeIsUntraced = FALSE
 (* From subtype ExprReservedIdRefTyp: *)


